clusterloader2/pkg/measurement/common/network/network_performance_measurement.go

```diff
@@ -0,0 +1,485 @@
+
+	customResourceGroup = "clusterloader.io"
+
+	customResourceKind = "NetworkTestRequest"
+
+	customResourceVersion = "v1alpha1"
+)
+
+const (
+	crdManifestFilePath = "$GOPATH/src/k8s.io/perf-tests/clusterloader2/pkg/measurement/common/network/manifests/*CustomResourceDefinition.yaml"
+
+	customResourceDefinitionGroup = "apiextensions.k8s.io"
+
+	customResourceDefinitionKind = "CustomResourceDefinition"
+
+	customResourceDefinitionName = "networktestrequests.clusterloader.io"
+)
+
+const (
+	clusterRoleBindingFilePath = "$GOPATH/src/k8s.io/perf-tests/clusterloader2/pkg/measurement/common/network/manifests/roleBinding.yaml"
+
+	rbacGroup = "rbac.authorization.k8s.io"
+
+	rbacKind = "ClusterRoleBinding"
+
+	rbacName = "networktestrequests-rbac"
+)
+
+func init() {
+	klog.Info("Registering Network Performance Measurement")
+	if err := measurement.Register(networkPerformanceMetricsName, createNetworkPerformanceMeasurement); err != nil {
+		klog.Fatalf("Cannot register %s: %v", networkPerformanceMetricsName, err)
+	}
+}
+
+func createNetworkPerformanceMeasurement() measurement.Measurement {
+	return &networkPerformanceMeasurement{}
+}
+
+type networkPerformanceMeasurement struct {
+	k8sClient       kubernetes.Interface
+	framework       *framework.Framework
+	podReplicas     int
+	namespace       string
+	podRatio        string
+	podRatioType    string
+	testDuration    int
+	protocol        string
+	templateMapping map[string]interface{}
+	// workerPodList stores list of podData for every worker node.
+	workerPodList map[string][]workerPodData
+
+	// metricVal stores MetricResponse received from worker node.
+	metricVal  [][]float64
+	metricLock sync.Mutex
+
+	podPairNumber int
+
+	podPairCh chan uniquePodPair
+
+	// startTimeStampForTestExecution is a futureTimeStamp sent to all pods to start
+	// sending the traffic at the same time.
+	startTimeStampForTestExecution int64
+
+	resultSummary testResultSummary
+}
+
+func (npm *networkPerformanceMeasurement) Execute(config *measurement.Config) ([]measurement.Summary, error) {
+	action, err := util.GetString(config.Params, "action")
+	if err != nil {
+		klog.Error("error starting action: ", err)
+		return nil, err
+	}
+
+	switch action {
+	case "start":
+		if err = npm.validate(config); err != nil {
+			klog.Error("error in validation action: ", err)
+			return nil, err
+		}
+		if err = npm.start(config); err != nil {
+			klog.Error("error starting measurement: ", err)
+			return nil, err
+		}
+	case "gather":
+		summary, err := npm.gather()
+		if err != nil && !errors.IsMetricViolationError(err) {
+			klog.Error("error gathering metrics: ", err)
+			return nil, err
+		}
+		klog.Info("metric: ", summary)
+		return []measurement.Summary{summary}, err
+	default:
+	}
+	return nil, nil
+}
+
+func (npm *networkPerformanceMeasurement) start(config *measurement.Config) error {
+	npm.initialize(config)
+
+	npm.namespace = netperfNamespace
```

Comment:
it seems there is no way to customize namespace - so let's remove this field from the type completely

Reply:
don't you use dynamic generated name like in the e2e framework?

Reply:
Yes , I have removed this field. I am using the const value directly.
@aojea : No we dont use dynamic generated namespaces names.